backend/lib/calculate/adequacy.py

```diff
@@ -126,6 +126,14 @@ def _get_locations_to_check_by_service_area(
     return locations_to_check_by_service_area
 
 
+@timed
+def _add_closest_provider_id(adequacies_response, location_mapping):
```

Comment:
You are adding more than one ID here. Open question: does this make more sense as a function that operates on a single `_find_closest_location` response or should it operate on the full list of them?